core/uemis-downloader.c

```diff
@@ -583,7 +593,7 @@ static bool uemis_get_answer(const char *path, char *request, int n_param_in,
 		if (import_thread_cancelled)
 			return false;
 		progress_bar_fraction = filenr / (double)UEMIS_MAX_FILES;
-		snprintf(fl, 13, "ANS%d.TXT", filenr - 1);
+		snprintf(fl, 13, "ANS%d.TXT", sanitize_filenr(filenr - 1));
```

Comment:
how about turning this in a utility function:
```
void write_filenr(char *buf, int bufsz, int filenr) 
{
    filenr = filenr < 0 ? 0 : filenr % 10000;
    snprintf(buf, bufsz, "ANS%d.TXT", filenr);    
}

// usage
write_filenr(fl, sizeof(fl), filenr - 1)
```

Reply:
Yes, fine by me. The question remains: do we actually want to code around compiler warnings?

Reply:
I'd like the `write_filenr()` helper better...